[Isaac Lim] iP

Isaaclks7 opened the pull request.

Isaaclks7 committed:
Rename Duke to Ducky

Isaaclks7 committed:
Add Greet.java

Isaaclks7 committed:
Add Echo in Greet.java

Isaaclks7 committed:
Add ability to store and list texts

Isaaclks7 committed:
Add Task.java

Isaaclks7 committed:
Add ability to mark tasks as done

Isaaclks7 committed:
Add printBorder method

Isaaclks7 committed:
Add Deadline.java, Todo.java, Event.java

Isaaclks7 committed:
Add Parser.java

To receive and split input string into the different commands required

Isaaclks7 committed:
Modify Greet.java, Task.java to track different tasks

Isaaclks7 committed:
Rename Greet.java to Ducky.java

Isaaclks7 committed:
Modify Ducky.java and Parser.java

Added methods to handle invalid or missing inputs

Isaaclks7 committed:
Modify response to invalid commands

Isaaclks7 committed:
Add input.txt, EXPECTED.TXT, runtest.bat for auto testing

Isaaclks7 committed:
Improve code readability using methods

xubin0 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,84 @@
+public class Parser {
+
+    public static class CommandResult {
```
It is confusing to have a class in a class

xubin0 commented on src/main/java/Ducky.java:
```diff
@@ -0,0 +1,153 @@
+
+        Scanner in = new Scanner(System.in);
+        String line;
+        Task[] taskList = new Task[100];
```
could this "100" be replaced with a constant instead?

Isaaclks7 replied:
Thanks for the suggestions!

xubin0 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,84 @@
+            eventStart = s.substring(s.indexOf("/from") + 6, s.indexOf("/to"));
+            eventEnd = s.substring(s.indexOf("/to") + 4);
+            break;
+        default:
```
is this default redundant?

xubin0 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+        return "";
+    }
+
+    public String getEventStart() {
```
this seemed redundant

xubin0 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+        return "";
+    }
+
+    public String getEventEnd() {
```
this seemed redundant

xubin0 reviewed (commented):
There were some redundancy of methods in parent class "Task"
Nested class was a little confusing

clj55 commented on src/main/java/Ducky.java:
```diff
@@ -31,90 +32,118 @@ public static void main(String[] args) {
+        printBorder();
+    }
 
+    private static void showValidCommands() {
```
very nice abstracting all these functions out, makes it more readable

clj55 commented on src/main/java/Parser.java:
```diff
@@ -54,6 +47,7 @@ public static CommandResult extract(String s) {
             return null;
         }
 
+        //Ensures valid command for adding a task
```
loving the comments, i cant see your whole code but can tell whats going on

clj55 commented on src/main/java/Ducky.java:
```diff
@@ -31,90 +32,118 @@ public static void main(String[] args) {
                 break;
             case "deadline":
-                taskList[taskCount] = new Deadline(result.description, result.deadline);
+                addDeadline(taskList, taskCount, result);
```
this SLAPs hard, v cool

Isaaclks7 replied:
Thanks!

clj55 commented on src/main/java/Event.java:
```diff
@@ -21,14 +21,6 @@ public String getEventEnd() {
         return eventEnd;
     }
 
-    public void setEventStart(String eventStart) {
```
why do you remove the setter functions?

clj55 reviewed (commented):
overall great code quality, just a nit in the Event class. I think the setter function isn't being used now but it could be useful for future iterations of your code where u might have to implement changing the start/end time.

Isaaclks7 committed:
Modified add task method in Ducky.java

Placed the different add task methods for each type of task in their respective classes for better organisation

Isaaclks7 committed:
Organise structure of Ducky.java, Task class and its subclasses

Isaaclks7 committed:
Replaced commandResult with Parser

Isaaclks7 committed:
Add DuckyException.java to handle invalid commands

Isaaclks7 committed:
Merge branch-Level-5 onto master. Added exceptions.

Isaaclks7 committed:
Created ducky package and placed task classes into task package

Isaaclks7 committed:
Replaced fixed array of tasks with array list

Isaaclks7 committed:
Update test inputs and outputs

Isaaclks7 committed:
Add exception handling for delete function

Isaaclks7 committed:
Add function to write added tasks to DuckyList.txt

Isaaclks7 committed:
Added TaskStorage to load existing list of tasks and append to the list when adding tasks

Isaaclks7 committed:
Add function to update data in txt file when updating task list

Isaaclks7 committed:
Merge branch-Level-6 to add delete task function

Isaaclks7 committed:
Merge with branch-Level-7 to add data storage functionalities

Isaaclks7 committed:
Added toString for each task type for simplicity

Isaaclks7 committed:
Edit path to data folder before packing as Jar

Isaaclks7 committed:
Creates data folder if does not exist

xenthm commented on src/main/java/ducky/DuckyException.java:
```diff
@@ -0,0 +1,43 @@
+        System.out.println("      5. Mark a task as done: mark <task index>");
+        System.out.println("      6. Unmark a task as undone: unmark <task index>");
+        System.out.println("      7. Delete a task: delete <task index>");
+        System.out.println("      8. End program: bye");
```
I like the use of indentations in your messages to make them more visually structured to users, but perhaps consider using a constant to represent it (i.e. `INDENT`) to avoid [magic numbers](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers).

xenthm commented on src/main/java/ducky/Parser.java:
```diff
@@ -0,0 +1,69 @@
+        try {
+            command = userInput.substring(0, userInput.indexOf(" "));
+            command = command.toLowerCase();
+        } catch (Exception e) {
```
Good use of `try`-`catch` blocks to handle exceptions! However, do consider using a more specific subclass of the generic `Exception` (maybe even creating your own) to provide more context and potentially make debugging easier.

xenthm commented on src/main/java/ducky/Parser.java:
```diff
@@ -0,0 +1,69 @@
+                return null;
+            }
+            break;
+        }
```
Friendly reminder to include the `default` case to [avoid unsafe shortcuts](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch).

xenthm commented on src/main/java/task/TaskStorage.java:
```diff
@@ -0,0 +1,113 @@
+                description = line.substring(line.indexOf("-") + 1);
+                taskList.add(new Todo(description, isDone));
+                break;
+            case "D":
+                    description = line.substring(line.indexOf("-") + 1, line.indexOf(" By: "));
+                    String deadline = line.substring(line.indexOf("By: ") + 4);
+                    taskList.add(new Deadline(description, isDone, deadline));
+                break;
```
Be careful with indentations in `switch` blocks! Lines 29-31 should be in line with `break;`

Isaaclks7 committed:
Created Command and Ui classes

Isaaclks7 committed:
Allow deadlines to convert yyyy-mm-dd to MMM dd yyyy format (Oct 10 2025)

Isaaclks7 committed:
Add function to Find tasks with an input key word

Isaaclks7 committed:
Added JavaDoc comments

Isaaclks7 committed:
Merge pull request branch-Level-8

Allow deadlines to convert yyyy-mm-dd to MMM dd yyyy format

Isaaclks7 committed:
Merge branch 'master' into branch-Level-9

Isaaclks7 committed:
Merge branch-A-JavaDoc onto master

Isaaclks7 committed:
Merge pull request branch-Level-9

Add function to Find tasks with an input key word

Isaaclks7 committed:
Merge pull request branch-A-JavaDoc

Added JavaDoc comments

Isaaclks7 committed:
Allow find function to be case-insensitive

Isaaclks7 committed:
Merge pull request #4 from Isaaclks7/branch-Level-9

Allow find function to be case-insensitive

Isaaclks7 committed:
Update input and expected test files

Isaaclks7 committed:
Update README.md

Isaaclks7 committed:
Added text file which stores tasks into gitignore

Isaaclks7 committed:
Merge branch 'master' of https://github.com/Isaaclks7/ip

Isaaclks7 committed:
Update docs/READE.md to show User Guide

Isaaclks7 committed:
Update README to show User Guide.

Isaaclks7 committed:
Updated format

Isaaclks7 committed:
Add contents page with links

Isaaclks7 committed:
Fixed bugs causing incorrect loading of existing tasks from txt file
